### DISCUSSION TITLE
Registers picker

### DISCUSSION THREAD

**Discussion by QiBaobin** [helix-editor/helix#10206] [Ideas]
Is it a good idea to implement a registers content picker? and override keys(enter, c-s, c-v) to support paste before/after and replace.
Current we can NOT reuse predefined register like `:`, `/`, `"`, `|` etc easily as it always select from the first item in those history.
To implement this picker, we can easily select items from all those registers and use them. 

If it makes sense, I can try to open a PR for this. Thanks!

**Comment by the-mikedavis** [comment]
For those, would it be enough to be able to select the last value in the register? I wonder if that could be a part of the `select_register` menu. A picker for them would seem not quite right to me since registers are a kind of key-value pair / dictionary and pickers are usually for sets of values / lists.

**Comment by QiBaobin** [comment]
> For those, would it be enough to be able to select the last value in the register? I wonder if that could be a part of the `select_register` menu. A picker for them would seem not quite right to me since registers are a kind of key-value pair / dictionary and pickers are usually for sets of values / lists.

That would make sense to me if we provide both select_register and select_register_last. As we still need use all of items for some commands.

**Comment by gibbz00** [comment]
> A picker for them would seem not quite right to me since registers are a kind of key-value pair / dictionary and pickers are usually for sets of values / lists.

Working on this right now! Actually, Registers are key-value pair (hash map), **but the values are basically `Vec<String>` with names attached to them**. All commands are stored in the Register named ":" and, I think, haven't looked so far yet, same holds true for past searches. See `registers.rs` and `prompt.rs` respectively. Or in other words, command history traversal (and probably past searches traversal) are in essence register pickers already :)
